docs: add changelog with notes for 1.6.0 (#734)
canonical/rockcraft: docs/reference/changelog.rst

Author: https://canonical-ubuntu-documentation-library--734.com.readthedocs.build/rockcraft/en/734/reference/changelog/

:tocdepth: 2

.. Check out the bottom of the page for the release template

Changelog
*********

1.6.0 (2024-Oct-17)
-------------------

Core
====

> - **Note**: This release of Rockcraft updates Pydantic, the internal library
>   used to process ``rockcraft.yaml`` files, to a new major version. This change
>   should not have any user-visible consequences but please report any issues at

Reviewer: ```suggestion
- Rockcraft 1.6.0 updates Pydantic, the internal library
  used to process ``rockcraft.yaml`` files, to a new major version. This change
  should not have any user-visible consequences, but please report any issues at
```

>   :literalref:`https://github.com/canonical/rockcraft/issues`.

Reviewer 2: ```suggestion
`on GitHub <https://github.com/canonical/rockcraft/issues>`_.
```

?

- Managed (non-destructive) runs now correctly fail if the build plan is empty.
  For example, trying to build a project with only ``amd64`` in its

>   ``platforms`` will now properly fail when buildings on a non-``amd64``

Author: I was trying to emphasize the fact that "failing to fail" was a bug here

Plugins
#######

``poetry``
""""""""""

- Add a new plugin for Python projects that use the `Poetry`_ build system.

Extensions
##########

fastapi-framework
"""""""""""""""""

- Add a new ``fastapi-framework`` extension for `FastAPI`_-based projects.

flask-framework
"""""""""""""""

- On ``bare``-based rocks, the extension now uses Chisel slices for the Python
  interpreter.
- Add support for ``ubuntu@24.04``.

django-framework
""""""""""""""""

- The ``django-framework`` extension is now stable and no longer requires the
  ``ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`` environment variable.
- Add support for ``ubuntu@24.04``.

go-framework
""""""""""""

- Add a new ``go-framework`` extension for Go-based projects.

Command line
============

- Improve reporting of builds that fail because they match multiple platforms.
- Improve error messages for invalid ``rockcraft.yaml`` files.
- The ``pack`` command now supports ``--shell`` and ``--shell-after``, and
  correctly handles cases where the packing itself fails and the ``--debug``
  argument is passed.
- The ``clean`` command now supports the ``--platform`` argument to filter which
  build environments to clean.

> - Positional arguments are now correctly display on the ``help`` output of

Reviewer 3: ```suggestion
- Positional arguments are now correctly displayed on the ``help`` output of
```

  commands.
- The terminal cursor is now hidden during execution.

Init
====

- Add new ``--profile`` argument options for ``go-framework`` and
  ``fastapi-framework``.

Documentation
=============

> - Add reference documentation for the new ``poetry`` plugin and the new

Reviewer 2: What about adding `:ref:` links for these?

  ``go-framework`` extension.
- Add a how-to guide on adding internal users to rocks.
- Improve the ``flask-framework`` tutorial page based on user feedback.
- Add tutorial pages for the existing ``django-framework`` and the new
  ``fastapi-framework`` extensions.

For a complete list of commits, check out the `1.6.0`_ release on GitHub.

.. _FastAPI:        https://fastapi.tiangolo.com
.. _Poetry:         https://python-poetry.org

.. _1.6.0:          https://github.com/canonical/rockcraft/releases/tag/1.6.0

> ..

Reviewer: I can think of arguments for putting it in either place, but FYI in Snapcraft we have the template at the top of the file.

Author: Right, I "took inspiration" from snapcraft's template. My thinking for putting the template at the bottom is that it's a long preamble that I have to scroll over when updating the document, but I could go either way. If we decide to put the template at the top that's fine with me.

Reviewer 2: I never really questioned it, but this gives me a weak preference for the template at the bottom.
